changelog/796.breaking.2.rst

```diff
@@ -0,0 +1,2 @@
+The ``State`` namedtuple was changed to the `~plasmapy.particles.IonicFraction`
+class.
```

Comment:
This entry should be deleted since it got renamed after that to `~plasampy.particles.IonicLevel` in #1046.

Reply:
The point of this is to document the changes in every PR contributed between releases, thus, I don't think it should be deleted.  It's an appropriate Traceback for us.

Reply:
Perhaps a compromise could be putting in a note saying that this PR was superseded by #1046?  

I tend to think of the changelog as a means to communicate with the end user about the differences between the current and previous release, whereas developers will probably search through pull requests or look through the commit log for the intermediate changes.  I was worried that this could be misleading, but a note would be enough to resolve that.

Reply:
> I tend to think of the changelog as a means to communicate with the end user about the differences between the current and previous release, whereas developers will probably search through pull requests or look through the commit log for the intermediate changes. I was worried that this could be misleading to the end user, but a note would be enough to resolve that.

I disagree.  A change log is for all users, pure users and contributors.  While PRs can be a good source for tracing back changes, they are often unorganized and not summarized well.  As a developer, I often go to the change log to get a brief summarized description of changes before spending the time to understand the often chaotic PR discussions.

I feel a change log should document all major steps (PRs) that lead to the next release, not a selectively edited change log for the release.  This improves traceback if it's ever needed.

> Perhaps a compromise could be putting in a note saying that this PR was superseded by #1046?

I don't feel that's really necessary.  A change log entry is a summary of changes at that point in development, it's ignorant of all future changes.  I don't want us to get in the habit of of going back in time to update change log entries because and new PR has changed things.  That would be a major headache!   I mean, I really hate how many changes I'm making in this PR but I'm discovering a lot off entries don't make sense a month after the fact.  Thus, I'm having to go back and make them more readable, but I'm not changing the intention of the change log.

Reply:
Sure, but in the bright future in which we've gotten better about writing good changelog entries, such changelog entry reviews should become less about fixing formatting and more about making these clear to end users. I'll add a note here to the tune of `(however, #number subsequently changed that to 'properlink.IonicLevel').`, with good formatting, of course.